client/src/client.rs

```diff
@@ -29,51 +29,63 @@ use crate::{
-    pub transaction_limits: TransactionLimits,
+    transaction_limits: TransactionLimits,
     /// Accounts keypair
-    pub key_pair: KeyPair,
+    key_pair: KeyPair,
     /// Transaction time to live in milliseconds
-    pub proposed_transaction_ttl_ms: u64,
+    proposed_transaction_ttl_ms: u64,
     /// Transaction status timeout
-    pub transaction_status_timeout: Duration,
+    transaction_status_timeout: Duration,
     /// Current account
-    pub account_id: AccountId,
+    account_id: AccountId,
     /// Http headers which will be appended to each request
-    pub headers: http_client::Headers,
+    headers: http_client::Headers,
     /// If `true` add nonce, which makes different hashes for
     /// transactions which occur repeatedly and/or simultaneously
-    pub add_transaction_nonce: bool,
+    add_transaction_nonce: bool,
 }
 
 /// Representation of `Iroha` client.
 impl Client {
-    /// Constructor for client
-    pub fn new(configuration: &Configuration) -> Self {
-        Self {
-            torii_url: configuration.torii_api_url.clone(),
-            telemetry_url: configuration.torii_telemetry_url.clone(),
-            transaction_limits: configuration.transaction_limits,
-            key_pair: KeyPair {
-                public_key: configuration.public_key.clone(),
-                private_key: configuration.private_key.clone(),
-            },
-            proposed_transaction_ttl_ms: configuration.transaction_time_to_live_ms,
-            transaction_status_timeout: Duration::from_millis(
-                configuration.transaction_status_timeout_ms,
-            ),
-            account_id: configuration.account_id.clone(),
-            headers: HashMap::default(),
-            add_transaction_nonce: configuration.add_transaction_nonce,
-        }
+    /// Constructor for client from configuration
+    ///
+    /// # Errors
+    /// Fails if configuration contains only `login` or only `password`.
+    /// It should contain both or none of them
+    pub fn new(configuration: &Configuration) -> Result<Self> {
+        Self::with_headers(configuration, http_client::Headers::default())
     }
 
-    /// Constructor for client
-    pub fn with_headers(configuration: &Configuration, headers: HashMap<String, String>) -> Self {
-        Self {
+    /// Constructor for client from configuration and headers
+    ///
+    /// *Authentication* header will be added, if `login` and `password` fields are presented
+    ///
+    /// # Errors
+    /// Fails if configuration contains only `login` or only `password`.
+    /// It should contain both or none of them
+    pub fn with_headers(
+        configuration: &Configuration,
+        mut headers: HashMap<String, String>,
+    ) -> Result<Self> {
+        match (&configuration.login, &configuration.password) {
+            (Some(login), Some(password)) => {
+                let credentials = format!("{}:{}", login, password);
+                let encoded = base64::encode(credentials);
```

Comment:
why does `base64` come all the way from `iroha_crypto` just to be used here? I think it would be fine to make it a dependency of `iroha_client` directly

Reply:
@appetrosyan suggested this approach. Well, it's done as a feature of `iroha_cryto`, so other crates don't get this library. But if any crate in future will need this, it will be easy to enable this feature.
Let's see what others think

Reply:
I actually suggested something closer to the first approach. You will raise a serialisation error when you can't find both the login and password. So you don't need an explicit check for whether you don't have either.

Reply:
While I think that  hiding it behind a feature flag is the right approach, it might be a premature optimization. When I made the suggestion, I assumed that Iroha will also need to process that information in order not to break, but since no changes are made beyond the client, we can do either.

Reply:
I'll wait for more responses here

Reply:
I should add that `base64` is encoding and not a form of encryption and if seen from that perspective shouldn't belong to `iroha_crypto`

Reply:
Ok, I'll move it to `client`